src/ccs/multifolding.rs

```diff
@@ -56,29 +58,51 @@ impl<G: Group> NIMFS<G> {
       ccs_mle,
       ck,
       lcccs,
+      transcript: TranscriptEngineTrait::new(b"NIMFS"),
     }
   }
 
   /// Initializes a NIMFS instance given the CCS of it and a first witness vector that satifies it.
   // XXX: This should probably return an error as we should check whether is satisfied or not.
-  pub fn init<R: RngCore>(mut rng: &mut R, ccs: CCS<G>, z: Vec<G::Scalar>) -> Self {
+  pub fn init(ccs: CCS<G>, z: Vec<G::Scalar>, seed: &'static [u8]) -> Self {
+    let mut transcript: G::TE = TranscriptEngineTrait::new(seed);
     let ccs_mle: Vec<MultilinearPolynomial<G::Scalar>> =
       ccs.M.iter().map(|matrix| matrix.to_mle()).collect();
+    // Add ccs circuit description to transcript. XXX: This does not need to be kept.
```

Comment:
Can you elaborate on this? Why do we add this, why does it no need to be kept and why do we keep it if it doesn't need to be kept?